src/ram/Expression.h

```diff
@@ -27,6 +27,7 @@ namespace souffle::ram {
 class Expression : public Node {
 public:
     Expression* cloning() const override = 0;
+    virtual ~Expression() = default;
```

Comment:
Why do we need destructors here? This seems unrelated?

Reply:
The COUNT nodes hold a ram::Expression* which under the hood can be a ram::SignedConstant or ram::StringConstant or ...

Since ram::Expression is a polymorphic type, it needs a virtual destructor, right?

Reply:
I don't believe so since the ram::Node has a virtual destructor. It should be alright.